python/grass/temporal/temporal_granularity.py

```diff
@@ -111,6 +112,81 @@ def check_granularity_string(granularity, temporal_type):
 ###############################################################################
 
 
+def get_time_tuple_function(maps):
+    """Helper function to return the appropriate function to get
+        time tuple from either TGIS DB rows or AbstractDataset object
+
+    :param maps: a list of AbstractDataset objects or database rows
+    :return: A function
+
+    .. code-block:: python
+
+        >>> from grass.temporal.abstract_map_dataset import AbstractMapDataset
+        >>> maps = AbstractMapDataset()
+        >>> get_time_tuple_function(maps).__name__
+        '_get_map_time_tuple'
+
+    """
+
+    def _get_map_time_tuple(map_object):
+        """Sub-function to return time tuple
+        from AbstractDataset object"""
+        if map_object.is_time_absolute():
+            time_tuple = map_object.get_absolute_time()
+        if map_object.is_time_relative():
+            time_tuple = map_object.get_relative_time()
+        return time_tuple[0:2]
+
+    def _get_row_time_tuple(db_table_row):
+        """Sub-function to return time tuple
+        from database row"""
+        return db_table_row["start_time"], db_table_row["end_time"]
+
+    # Check if input is list of MapDataset objects or SQLite rows
+    if issubclass(maps[0].__class__, AbstractMapDataset):
+        return _get_map_time_tuple
+    else:
+        return _get_row_time_tuple
+
+
+def _is_after(start, start1, end1):
+    """Helper function that checks if start timestamp is
+    temporaly after the start1 and end1, where start1 and end1
```

Comment:
```suggestion
    temporally after the start1 and end1, where start1 and end1
```

Reply:
Fixed in [04d4778](https://github.com/OSGeo/grass/pull/3350/commits/04d4778d32e34a4c3842187fd22bb7fbc4a8d34e)